test/unit/data_hygiene/organisation_reslugger_test.rb

```diff
@@ -30,47 +30,33 @@ def teardown
     end
 
     test "publishes to Publishing API with the new slug and redirects the old" do
-      content_item = PublishingApiPresenters.presenter_for(@organisation)
-      old_base_path = @organisation.search_link
       new_base_path = "#{base_path}/corrected-slug"
 
+      content_item = PublishingApiPresenters.presenter_for(@organisation)
       content = content_item.content
       content[:base_path] = new_base_path
       content[:routes][0][:path] = new_base_path
-
       content_item.stubs(content: content)
 
-      redirect_uuid = SecureRandom.uuid
-      SecureRandom.stubs(uuid: redirect_uuid)
-
-      redirects = [
-        { path: old_base_path, type: "exact", destination: new_base_path },
-      ]
-
-      if @organisation.is_a? Organisation
-        redirects << { path: (old_base_path + ".atom"),
-                       type: "exact",
-                       destination: (new_base_path + ".atom") }
-      end
-
-      redirect_item = PublishingApiPresenters::Redirect.new(old_base_path, redirects)
-
       expected_publish_requests = [
         stub_publishing_api_put_content(content_item.content_id, content_item.content),
         stub_publishing_api_patch_links(content_item.content_id, links: content_item.links),
         stub_publishing_api_publish(content_item.content_id, locale: 'en', update_type: 'major')
       ]
 
-      expected_redirect_requests = [
-        stub_publishing_api_put_content(redirect_item.content_id, redirect_item.content),
-        stub_publishing_api_patch_links(redirect_item.content_id, links: redirect_item.links),
-        stub_publishing_api_publish(redirect_item.content_id, locale: 'en', update_type: 'major')
-      ]
+      request = stub_publishing_api_unpublish(
+        @organisation.content_id,
```

Comment:
Good point, if the slug is updated before the redirect is issued the content item published by the slug update will be redirected, this isn't right. The redirect unpublishing needs to happen before the slug update. This runs the risk of redirecting then failing to update (the slug) leaving a broken redirect on the router though...